Enhanced boneyard

perlmonger42 opened the pull request:
All game pieces are represented in `div.dead-pieces`. 

- Dead pieces are rendered as before, except that they now have a mouse-over tool tip that reads, e.g., "dead red-horse". 
- Pieces that are currently in play (i.e., face up on the board) are rendered in the boneyard dimmed-out (opacity:20%) with a tooltip like "live black-canon". 
- Pieces that have not yet been turned up are rendered in the boneyard as a dimmed-out green disc with a tooltip like "unborn red-king".

I hadn't seen https://github.com/arbrown/pao/issues/38#issuecomment-632743229 until after I'd coded this up, so it's only similar to that picture. I'll try to CSS unborn pieces into that look, and report back.

arbrown commented:
Nice!  I think I would prefer the look from the comment in #38.  Maybe it's just because I'm used to it, but I like the idea of live pieces looking like they do on the board with dead pieces being the 'dim' ones.

Is there any way to achieve a similar effect besides opacity?  I can't say I really like the look of semi-transparent pieces showing the background through.

![2020-05-30 17_24_17-Ban Qi](https://user-images.githubusercontent.com/498727/83340932-ffe3bd80-a29a-11ea-9ffd-c5502c3f42c7.png)

I'll play around with some of the css classes and see if I can come up with any alternatives.

perlmonger42 commented:
I managed to get the green tint. I wish the CSS weren't so bulky, though. And I agree with you about the undesirability of having the board background showing through.

I don't like the look of it, though. 
![Screen Shot 2020-05-30 at 5 46 23 PM](https://user-images.githubusercontent.com/335065/83341171-8bf6e480-a29d-11ea-8cb9-e5944feaa1fc.png)

Should we try for exactly the #38 pic? I.e., in-play pieces are rendered normally, face-down with a green tint, and dead pieces with a gray shading?

arbrown commented:
What do you think about this?
```
.dead-pieces .banqi-square.dead::before {
    content: "";
    background-color: black;
    opacity: .4;
    width:inherit;
    height:inherit;
    position: fixed;
}

.dead-pieces .banqi-square.unborn::before {
    content: "";
    background-color: green;
    opacity: .4;
    width:inherit;
    height:inherit;
    position: fixed;
}
```
![2020-05-30 18_04_18-Ban Qi](https://user-images.githubusercontent.com/498727/83341390-1f311980-a2a0-11ea-80a8-6bdf68ac3cab.png)

This gets rid of the css styles you had for .unborn and .live.

I'd like to somehow crop the pseudo elements to be circles, which is probably possible.  I might look at it more if you think this is an ok solution.

perlmonger42 commented:
Yes, I think that looks reasonable, especially if you can get circles. That
example has no dead pieces, right?

Is the `::before` a dodge that allows you to get it to apply both `.unborn`
and (e.g.) `.banqi-square.red-horse` styles?

On Sat, May 30, 2020 at 6:07 PM Drew Brown <notifications@github.com> wrote:

> What do you think about this?
>
> .dead-pieces .banqi-square.dead::before {
>     content: "";
>     background-color: black;
>     opacity: .4;
>     width:inherit;
>     height:inherit;
>     position: fixed;
> }
>
> .dead-pieces .banqi-square.unborn::before {
>     content: "";
>     background-color: green;
>     opacity: .4;
>     width:inherit;
>     height:inherit;
>     position: fixed;
> }
>
> [image: 2020-05-30 18_04_18-Ban Qi]
> <https://user-images.githubusercontent.com/498727/83341390-1f311980-a2a0-11ea-80a8-6bdf68ac3cab.png>
>
> This gets rid of the css styles you had for .unborn and .live.
>
> I'd like to somehow crop the pseudo elements to be circles, which is
> probably possible. I might look at it more if you think this is an ok
> solution.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/arbrown/pao/pull/52#issuecomment-636400763>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACRZWKPGQPAMFYV6ALXPPLRUGNS5ANCNFSM4NO4LEWA>
> .
>

arbrown commented on client/content/banqi.css:
```diff
@@ -105,14 +105,21 @@ html:before {
   margin:2px;
   display:inline-block;
 }
+.dead-pieces .banqi-square.live {
+    opacity: 20%;
+}
+.dead-pieces .banqi-square.unborn {
+  opacity: 20%;
+  background-image: url("/images/unflipped-piece.svg");
+}
```
```suggestion

```

arbrown replied:
I think this change will get closer to the image, and it gets rid of the distracting opacity issues.  It even crops the effect to the circle of the piece.  Thoughts?

perlmonger42 replied:
I think that looks good. Here's what I'm seeing:
![Screen Shot 2020-05-30 at 6 22 58 PM](https://user-images.githubusercontent.com/335065/83341625-a3849c00-a2a2-11ea-8bdd-38467fd20175.png)

perlmonger42 commented:
So, with your change, I'm now seeing this:
![Screen Shot 2020-05-30 at 6 48 51 PM](https://user-images.githubusercontent.com/335065/83341940-47237b80-a2a6-11ea-9ae9-6c68d72af501.png)

I just wish the dead pieces stood out a little more relative to unflipped pieces. I even tried a big red x. Which may be overkill, but I kinda like. I just can't get 'em centered:
![Screen Shot 2020-05-30 at 6 50 41 PM](https://user-images.githubusercontent.com/335065/83341953-7a660a80-a2a6-11ea-8ca2-7759ada3376f.png)

arbrown commented on client/content/banqi.css:
```diff
@@ -105,14 +105,33 @@ html:before {
+    content: "";
+    background-color: black;
+    opacity: 40%;
+    border-radius: ;
```
```suggestion
    border-radius: 50%;
```

Somehow this didn't merge correctly.

arbrown commented:
I merged the changes in, but we can definitely continue this discussion here or elsewhere.  I tested bumping up the opacity on the tint coloring to 60% and it makes the dead pieces stand out a little.  I like the 'X' idea as well, as long as it doesn't obscure the logogram like @walrii mentioned in the original request #38.
